### PR TITLE
Pass on architecture

### PR DESCRIPTION
**Pull request by chris-wood**
cc @tfpauly

### CODE REVIEW

File: draft-ietf-privacypass-architecture.md [hunk 1]
```diff
@@ -198,30 +198,40 @@ The end-to-end flow for Privacy Pass involves three different types of
...
 - Redemption context: The interactions and set of information shared
-between the Client and Origin.
+between the Client and Origin. This context includes all information
+associated with redemption, such as the timestamp of the event, Client
+visible information (including the IP address), and the Origin name.
 - Issuance context: The interactions and set of information shared
-between the Client, Attester, and Issuer.
+between the Client, Attester, and Issuer. This context includes all
+information associated with issuance, such as the timestamp of the event,
+any Client visibile information (including the IP address), and the
```
Comment (tfpauly) [inline draft-ietf-privacypass-architecture.md:207]:
```suggestion
any Client visible information (including the IP address), and the
```

File: draft-ietf-privacypass-architecture.md [hunk 2]
```diff
@@ -198,30 +198,40 @@ The end-to-end flow for Privacy Pass involves three different types of
...
+types of unlinkability:
+
+1. Origin-Client unlinkability. This means that given two redemption contexts,
+the Origin cannot determine if both redemption contexts correspond to the same
+Client or two different Clients. Informally, this means that a Client in a
+redemption context is indistinguishable from any other Client that might use
+the same redemption context. The set of Clients that share the same redemption
+context is referred to as a redemption anonymity set.
+2. Issuer-Client unlinkability. This is similar to Origin-Client unlinkability
+in that a Client in an issuer context is indistingusihable from any other
```
Comment (tfpauly) [inline draft-ietf-privacypass-architecture.md:227]:
```suggestion
in that a Client in an issuer context is indistinguishable from any other
```

Comment (chris-wood) [inline draft-ietf-privacypass-architecture.md:227]:
```suggestion
in that a Client in an issuer context is indistinguishable from any other
```

File: draft-ietf-privacypass-architecture.md [hunk 3]
```diff
@@ -697,12 +697,12 @@ Origin-Client, Issuer-Client, and Attester-Origin unlinkability.
...
 unlinkable. For example, consider two different deployments, one wherein
-there exists a redemption context anonymity set of size two and another
-wherein there redemption context anonymity set of size 2<sup>32</sup>. Although
-Origin-Client and Issuer-Client unlinkabiity guarantees that Origin and Issuer
-cannot link any two requests to the same Client based on these contexts,
-respectively, the probability of determining the "true" Client is higher the
-smaller these sets become.
+there exists a redemption anonymity set of size two and another
+wherein there redemption anonymity set of size 2<sup>32</sup>. Although
+Origin-Client unlinkabiity guarantees that tje Origin cannot link any two
```
Comment (tfpauly) [inline draft-ietf-privacypass-architecture.md:702]:
```suggestion
Origin-Client unlinkabiity guarantees that the Origin cannot link any two
```

File: draft-ietf-privacypass-architecture.md [hunk 4]
```diff
@@ -198,30 +198,40 @@ The end-to-end flow for Privacy Pass involves three different types of
...
+between the Client, Attester, and Issuer. This context includes all
+information associated with issuance, such as the timestamp of the event,
+any Client visibile information (including the IP address), and the
+Origin name (if revealed during issuance).
 - Attestation context: The interactions and set of information shared between
 the Client and Attester only, for the purposes of attesting the vailidity of
-the Client.
+the Client. This context includes all information associated with attestation,
+such as the timestamp of the event and any Client visibile information,
+including the IP address or other information specific to the type of
```
Comment (dvorak42) [inline draft-ietf-privacypass-architecture.md:213]: the wording here seems a little odd, maybe implying IP address and other information is necessarily visible?

**Review by tfpauly — APPROVED** [review]

File: draft-ietf-privacypass-architecture.md [hunk 5]
```diff
@@ -198,30 +198,40 @@ The end-to-end flow for Privacy Pass involves three different types of
...
+information associated with issuance, such as the timestamp of the event,
+any Client visible information (including the IP address), and the
+Origin name (if revealed during issuance).
 - Attestation context: The interactions and set of information shared between
 the Client and Attester only, for the purposes of attesting the vailidity of
-the Client.
+the Client. This context includes all information associated with attestation,
+such as the timestamp of the event and any Client visibile information,
+including the IP address or other information specific to the type of
+attestation done.
```
Comment (chris-wood) [inline draft-ietf-privacypass-architecture.md:214]:
```suggestion
including information needed for the attestation procedure to complete.
```

File: draft-ietf-privacypass-architecture.md (hunk 4 above)
Comment (chris-wood) [reply draft-ietf-privacypass-architecture.md]: Rephrased slightly to focus just on attestation information.

File: draft-ietf-privacypass-architecture.md [hunk 6]
```diff
@@ -330,23 +340,23 @@ Each issuance protocol may be different, e.g., in the number and types of
...
+Issuer, or other parties that produces a valid token bound to the Client's
 private input, subject to the following security requirements.
 
 1. Unconditional input secrecy. The issuance protocol MUST NOT reveal anything
 about the Client's private input, including the challenge and nonce, to the
 Attester or Issuer, regardless of the hardness assumptions of the underlying
 cryptographic protocol(s). The issuance protocol can reveal the Issuer public
 key for the purposes of determining which private key to use in producing the
-token.
+token. This property is sometimes also referred to as blindness.
```
Comment (dvorak42) [inline draft-ietf-privacypass-architecture.md:358]: Do we need to explicitly refer to this property as "blindness" or just leave it as Unconditional input secrecy?

File: draft-ietf-privacypass-architecture.md (hunk 3 above)
Comment (dvorak42) [inline draft-ietf-privacypass-architecture.md]: nit: the

Comment (smhendrickson) [inline draft-ietf-privacypass-architecture.md]: sp: the

**Review by smhendrickson — APPROVED** [review]

File: draft-ietf-privacypass-architecture.md (hunk 6 above)
Comment (chris-wood) [reply draft-ietf-privacypass-architecture.md:358]: Let's leave this as-is.

**Pull request merged**